### PR TITLE
Wrzucam poradnik z fluttera i sqlite

### PR DESCRIPTION
**Pull request by tobiaszciesielski**

### CODE REVIEW

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+``` dart
+import 'package:sqflite/sqflite.dart';              // Database, openDatabase()
+import 'package:path/path.dart';  	            // join()
+import 'package:path_provider/path_provider.dart';  // getApplicationDocumentsDirectory()
+import 'dart:io'; 				    // Diretory
+```
+---
+### 🚀 Krok 5. CRUD
+
+Stworzymy teraz funkcje do tworzenia, pobierania, aktualizowania i usuwania studentów. Należy też dodać model` Student `do naszego pliku `database.dart` .
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
Stworzymy teraz funkcje do tworzenia, pobierania, aktualizowania i usuwania studentów. Należy też dodać model `Student` do naszego pliku `database.dart` .
```

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+
+1. Pobieranie studentów lub studenta po id
+```dart
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md:192]: I've changed it to
```suggestion
1. Pobieranie studentów lub studenta po id

```dart

// ...
```
and it's parsed properly

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+  - "mobile"
+description: "Tworzenie bazodanowej aplikacji mobilnej krok po kroku z użyciem flutter i SQLite"
+---
+
+W tym poradniku przestawiam, jak napisać prostą aplikację do dodawania i usuwania studentów za pomocą interfejsu użytkownika w bazie danych. 
+
+---
+
+💡 Zanim przeczytasz upewnij się, że: 
+- 🔗Posiadasz zainstalowany [flutter](https://flutter.dev/docs/get-started/install),
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
- 🔗 Posiadasz zainstalowany [flutter](https://flutter.dev/docs/get-started/install),
```
let's have a space for that and subsequent icons

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+slug: "/posts/sqlite-flutter/"
+category: "mobile"
+tags:
+  - "sqlite"
+  - "flutter"
+  - "mobile"
+description: "Tworzenie bazodanowej aplikacji mobilnej krok po kroku z użyciem flutter i SQLite"
+---
+
+W tym poradniku przestawiam, jak napisać prostą aplikację do dodawania i usuwania studentów za pomocą interfejsu użytkownika w bazie danych.
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
W tym poradniku przedstawiam, jak napisać prostą aplikację do dodawania i usuwania studentów za pomocą interfejsu użytkownika w bazie danych. 
```

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+  
+  Student({  
+    this.id,  
+    this.firstName,  
+    this.lastName,  
+    this.grade  
+  });  
+}
+```
+To nie koniec. SQLite z naszą aplikacją wymienia się danymi w postaci [Mapy](https://www.tutorialspoint.com/dart_programming/dart_programming_map.htm) . Aby sprawnie przechodzić z instancji klasy na mapę i odwrotnie należy zaimplementować odpowiednie do tego metody.
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
To nie koniec. SQLite z naszą aplikacją wymienia się danymi w postaci [Mapy](https://www.tutorialspoint.com/dart_programming/dart_programming_map.htm). Aby sprawnie przechodzić z instancji klasy na mapę i odwrotnie należy zaimplementować odpowiednie do tego metody.
```

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+Zmapujemy ciąg znaków na dynamiczny typ danych ponieważ posiadamy różne rodzaje danych w modelu `Map<String, dynamic>`.
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
Zmapujemy ciąg znaków na dynamiczny typ danych, ponieważ posiadamy różne rodzaje danych w modelu `Map<String, dynamic>`.
```

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+}
+```
+Zauważ, że konstruktor klasy Student `fromMap` posiada słowo kluczowe `factory` (tak zwany *factory constructor*) dzięki któremu możemy obsłużyć logikę tworzenia instancji, której nie jest w stanie obsłużyć lista inicjalizacyjna. 
+
+🔗 Więcej o *factory consturctor* na [dart.dev](https://dart.dev/guides/language/language-tour#factory-constructors) oraz [stackoverflow](https://stackoverflow.com/questions/53886304/understanding-factory-constructor-code-example-dart).
+
+---
+### 📊 Krok 4. DatabaseProvider
+
+Pora zadbać o inicjalizację naszej bazy danych. Skorzystamy z wzorca [Singleton](https://refactoring.guru/pl/design-patterns/singleton) dzięki któremu obiekt `DatabaseProvider`  będzie jedynym tego rodzaju obiektem w naszej aplikacji. Taką logikę uzyskujemy za pomocą pola `static` instancji klasy oraz  prywatnego konstruktora. Dzięki temu instancja istnieje cały czas a prywatny konstruktor uniemożliwia stworzenia kolejnego obiektu z zewnątrz.
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
Pora zadbać o inicjalizację naszej bazy danych. Skorzystamy z wzorca [Singleton](https://refactoring.guru/pl/design-patterns/singleton) dzięki któremu obiekt `DatabaseProvider`  będzie jedynym tego rodzaju obiektem w naszej aplikacji. Taką logikę uzyskujemy za pomocą pola `static` instancji klasy oraz  prywatnego konstruktora. Dzięki temu instancja istnieje cały czas, a prywatny konstruktor uniemożliwia stworzenia kolejnego obiektu z zewnątrz. 
```

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+  
+  // static instance  
+  static final DatabaseProvider db = DatabaseProvider.internal();  
+  
+  // SQLite database  
+  Database _database;  
+}
+```
+
+Teraz potrzebujemy funkcji, która będzie zwracała nam połączenie z bazą danych lub tworzyła je jeżeli jeszcze nie zostało ustanowione.
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
Teraz potrzebujemy funkcji, która będzie zwracała nam połączenie z bazą danych lub tworzyła je, jeżeli jeszcze nie zostało ustanowione. 
```

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+    final tmpList = await DatabaseProvider.db.getAllStudentDrivers();  
+    setState(() {  
+      isLoading = false;  
+      studentsList = tmpList;  
+    });  
+  }  
+}
+```
+
+Struktura Widgetów naszej aplikacji aplikacji będzie wyglądała następująco.
```
Comment (roertbb) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
Struktura Widgetów naszej aplikacji będzie wyglądała następująco. 
```

**Review by roertbb — APPROVED** [review]
Great article! 👏 Even as non-flutter dev I was able to grasp the context - leaving comments with some minor tweaks like typos, possible punctation improvements and so on. Let's fix them, adjust the date and we should be ready to merge that!

File: content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md [hunk 10]
```diff
@@ -0,0 +1,427 @@
+---
+title: SQLite i flutter - z czym to się je.
+date: "2020-11-02T17:08:00.169Z"
```
Comment (tobiaszciesielski) [inline content/posts/2020-11-02--SQLite-i-flutter---z-czym-to się-je.md]:
```suggestion
date: "2020-11-06T20:28:00.169Z"
```

**Comment by tobiaszciesielski** [comment]
> 
> 
> Great article! 👏 Even as non-flutter dev I was able to grasp the context - leaving comments with some minor tweaks like typos, possible punctation improvements and so on. Let's fix them, adjust the date and we should be ready to merge that!

I really appreciate your words. I hope this artice will be helpful for begginers.

**Comment by roertbb** [comment]
Definitely! I'm not certain if you have some flow related to posting/frequency. Big ✅ from me - @marcinlawnik do the honors!

**Pull request merged**